core/templates/tests/unit-test-utils.ts

```diff
@@ -0,0 +1,155 @@
+ * @param {bool} allSpace - A bool to define whether to replace new line
+ *   with a space or empty string.
+ */
+export const multiTrim = (
+    text: string | null | undefined, allSpace = false): string => {
+  // eslint-disable-next-line eqeqeq
+  if (typeof text == 'string') {
+    const repl = allSpace ? '' : ' ';
+    return text.replace(/\n/g, '').replace(/\s+/g, repl).trim();
+  }
+  throw new Error('Argument can not be undefined.');
+};
+
+/**
+ * And upgraded angularjs module.
+ * @param {PlatformRef} platform - The platform reference for bootstraping.
+ * @param {Type<{}>} Ng2Module - Angular module to bootstrap.
+ * @param {Element} element - An html element.
+ * @param {angular.IModule} ng1Module - The angularjs module to be included
+ *  in angular zone.
+ */
+export const bootstrap = (
+    platform: PlatformRef,
+    Ng2Module: Type<{}>,
+    element: Element,
+    ng1Module: angular.IModule
+): Promise<UpgradeModule> => {
+// We bootstrap the Angular module first; then when it is ready (async) we
+// bootstrap the AngularJS module on the bootstrap element (also ensuring that
+// AngularJS errors will fail the test).
+  return platform.bootstrapModule(Ng2Module).then(ref => {
+    const ngZone = ref.injector.get<NgZone>(NgZone);
+    const upgrade = ref.injector.get(UpgradeModule);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const failHardModule: any = ($provide) => {
+      $provide.value('$exceptionHandler', (err) => {
+        throw err;
+      });
+      return '';
+    };
+
+    // The `bootstrap()` helper is used for convenience in tests, so that we
+    // don't have to inject and call `upgrade.bootstrap()` on every Angular
+    // module. In order to closer emulate what happens in real application,
+    // ensure AngularJS is bootstrapped inside the Angular zone.
+    ngZone.run(() => upgrade.bootstrap(
+      element, [failHardModule, ng1Module.name]));
+
+    return upgrade;
+  });
+};
+
+/**
+ * A utility function to get coverage of the upgraded component class.
+ * @param {string} kebabCaseName - Name of the upgraded component in kebab-case.
+ * @param {string} camelCaseName - Name of the upgraded component in camelCase.
+ * @param {unkownn} upgradedComponentTypes - An array consisting of only
+ *   one element. That element is the type of the upgraded component.
+ * @param {boolean} focus - To run this test in fdescribe.
+ */
+export const testUpgradeComponent = (
+    kebabCaseName: string,
+    camelCaseName: string,
+    upgradedComponentTypes: unknown,
+    focus?: boolean): void => {
+  let template = '<' + kebabCaseName + '></' + kebabCaseName + '>';
+  @Component({
+    selector: 'mock-comp',
+    template: template
+  })
+  class MockComponent {}
+  const test = () => {
+    beforeEach(() => destroyPlatform());
+    afterEach(() => destroyPlatform());
+    it('should cover the lines', async(() => {
+      const ng1Component = {template: 'Hello, Angular!'};
```

Comment:
Maybe add an explanation for 'Hello Angular'. Like why it is needed probably

Reply:
Done!